src/screens/PushNotification.js

```diff
@@ -0,0 +1,210 @@
+import React from 'react';
+import {
+  StyleSheet,
+  SafeAreaView,
+  Switch,
+  Image,
+} from 'react-native';
+import { connect } from 'react-redux';
+import { isEqual } from 'lodash';
+import HathorHeader from '../components/HathorHeader';
+import { HathorList, ListItem } from '../components/HathorList';
+import ActionModal from '../components/ActionModal';
+import { isEnablingFeature } from '../utils';
+import FeedbackModal from '../components/FeedbackModal';
+import errorIcon from '../assets/images/icErrorBig.png';
+import { STORE, pushNotificationKey } from '../constants';
+import { pushApiReady, pushFirstTimeRegistration, pushUpdateRequested } from '../actions';
+import { PUSH_API_STATUS } from '../sagas/pushNotification';
+
+const getPushNotificationSettings = (pushNotification) => {
+  const { enabled, showAmountEnabled } = pushNotification;
+  return {
+    enabled,
+    showAmountEnabled
+  };
+};
+
+// eslint-disable-next-line max-len
+const hasApiStatusFailed = (pushNotification) => pushNotification.apiStatus === PUSH_API_STATUS.FAILED;
+
+/**
+ * wallet {Object} wallet at user's device
+ */
+const mapInvoiceStateToProps = (state) => ({
+  wallet: state.wallet,
+  pushNotification: {
+    ...state.pushNotification,
+    ...(state.pushNotification.noChange && STORE.getItem(pushNotificationKey.settings)),
+    hasPushApiFailed: hasApiStatusFailed(state.pushNotification),
+  },
+});
+
+const mapInvoiceDispatchToProps = (dispatch) => ({
+  pushApiReady: () => dispatch(pushApiReady()),
+  pushFirstTimeRegistration: (deviceId) => dispatch(pushFirstTimeRegistration({ deviceId })),
+  pushUpdateRequested: (settings) => dispatch(pushUpdateRequested(settings)),
+});
+
+class PushNotification extends React.Component {
```

Comment:
Should we use function components instead of class components so we can slowly transition to more modern React?

Reply:
I'm not aware of it. Can you give some materials to read?